packages/headless/src/features/commerce/context/context-slice.ts

```diff
@@ -0,0 +1,32 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {getContextInitialState} from './context-state';
+import {setClientId, setContext, setCurrency, setLanguage, setTrackingId, setUser, setView} from './context-actions';
+
+export const contextReducer = createReducer(
+  getContextInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(setContext, (_, {payload}) => {
+        return payload
+      })
+      .addCase(setTrackingId, (state, {payload}) => {
```

Comment:
We could have a single `setContext` action.

And in `headless-context.ts`: 

```typescript
setTrackingId: (trackingId: string) => dispatch(setContext({...getState().commerceContext, trackingId})),
```

Reply:
Are there best practices, or guidelines for when to create an action, and when to reuse an existing one?

I'll make the change to use `setContext` for all 4 top-level `string` properties (`trackingId`, `language`, `currency`, `userId`) in the meantime.

Reply:
> Are there best practices, or guidelines for when to create an action, and when to reuse an existing one?

It's generally better to limit the amount of symbols that gets exported publicly from headless (just to limit the amount of potential breaking change to maintain on our side down the line). And all actions are exported publicly.

In this case I don't think there's much value added to have one action for every single property.
It would be different if we needed to do manipulation/verification/validation etc.. before setting a value in the store.